test/e2e/functional/tests/e2e.go

```diff
@@ -354,6 +365,826 @@ var _ = Describe("Ingress Node Firewall", func() {
+			Expect(err).ShouldNot(HaveOccurred())
+			//wait for daemonset to be rolled out
+			infDaemonSet := &appsv1.DaemonSet{}
+			err = daemonset.WaitForDaemonSetReady(testclient.Client, infDaemonSet, OperatorNameSpace,
+				"ingress-node-firewall-daemon", retryInterval, timeout)
+			Expect(err).ShouldNot(HaveOccurred())
+			//setup client pod which we always need
+			client = pods.GetNetcatClientPod(clientPodName, OperatorNameSpace, podLabel)
+			client, err = pods.EnsurePodRunning(testclient.Client, client, OperatorNameSpace, retryInterval, timeout)
+			Expect(err).ShouldNot(HaveOccurred())
+			server = pods.GetNetcatServerPod(serverPodName, OperatorNameSpace, serverListenPort, podLabel,
+				ingressnodefwv1alpha1.ProtocolTypeTCP)
+			server, err = pods.EnsurePodRunning(testclient.Client, server, OperatorNameSpace, retryInterval, timeout)
+			Expect(err).Should(BeNil())
+			inf = &ingressnodefwv1alpha1.IngressNodeFirewall{}
+			inf.SetName("e2e-inf-icmp-rules")
+			infwutils.DefineWithWorkerNodeSelector(inf)
+			infwutils.DefineWithInterface(inf, testInterface)
+			infwutils.AppendIngress(inf, fmt.Sprintf("%s/32", client.Status.PodIP),
+				infwutils.GetICMPV4Rule(1, 0, 8, ingressnodefwv1alpha1.IngressNodeFirewallDeny))
+			Expect(infwutils.CreateIngressNodeFirewall(testclient.Client, inf, timeout)).ShouldNot(HaveOccurred())
+			// TODO(mk): remove this sleep when we have a way to detect INF application is completed. Currently, we have no
+			// transition or pending state in the INF obj.
+			time.Sleep(200 * time.Millisecond)
+		})
+
+		AfterEach(func() {
+			pods.DeletePod(testclient.Client, client, timeout)
+			pods.DeletePod(testclient.Client, server, timeout)
+			infwutils.DeleteIngressNodeFirewall(testclient.Client, inf, timeout)
+			infwutils.DeleteINFConfig(config)
+		})
+
+		It("block echo replies", func() {
+			Eventually(func() bool {
+				_, stdErr, err := pods.Ping(testclient.Client, client, server.Status.PodIP, timeout)
+				if stdErr != "" || err != nil {
+					return true
+				}
+				return false
+			}, timeout, retryInterval)
+		})
+
+		It("allows connectivity when updated to allow echo replies", func() {
+			Eventually(func() bool {
+				inf, err = infwutils.GetIngressNodeFirewall(testclient.Client, inf.Name, inf.Namespace, timeout)
+				if err != nil {
+					return false
+				}
+				// redefine to ICMP type destination unreachable (3) and ICMP code net is unreachable (0)
+				infwutils.RedefineIngressICMPV4Rule(inf, 0, 0, 3, 0)
+				err = infwutils.UpdateIngressNodeFirewall(testclient.Client, inf, timeout)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, retryInterval)
+			Eventually(func() bool {
+				_, stdErr, err := pods.Ping(testclient.Client, client, server.Status.PodIP, timeout)
+				if stdErr != "" || err != nil {
+					return false
+				}
+				return true
+			}, timeout, retryInterval)
+		})
+	})
+
```

Comment:
see the above TCP/UDP/SCTP will be much cleaner if its a table which takes rule, protocol and expected results and even the matching string in the event logs for each to just one function